lib/modules/dosomething/dosomething_campaign/dosomething_campaign.theme.inc

```diff
@@ -213,17 +213,17 @@ function dosomething_campaign_preprocess_action_page(&$vars, &$wrapper) {
   // 1) Materials
 
   // Build the unique share link.
-  $share_path = 'user/' . $vars['user']->uid;
+  $share_path = 'user/' . $vars['user']->uuid;
```

Comment:
`uuid` is actually not the same as NS id. it's a drupaly thing
`field_data_field_northstar_id` is where you wanna grab the right id